pkgs/development/libraries/tbb/default.nix

```diff
@@ -51,7 +51,7 @@ stdenv.mkDerivation rec {
 
   makeFlags = lib.optionals stdenv.cc.isClang [
     "compiler=clang"
-  ];
+  ] ++ (lib.optional ((stdenv.buildPlatform != stdenv.hostPlatform) && stdenv.hostPlatform.isAarch64) "arch=arm64");
```

Comment:
```suggestion
  ] ++ [ "arch=${stdenv.hostPlatform.uname.processor}" ];
```

Reply:
This doesn't work generally, since the names don't seem to match the linuxArch values. For example, tbb wants "intel64" instead of "x86_64"

Reply:
Then we can hardcode an tbb specific mapping. Also I noticed that our tbb is out of date, would you like to take the chance to upgrade it as well?

Reply:
It's not obvious to me where to get these values. Grepping the tbb source didn't reveal as much as I expected. And I don't have a way to test that the other ones besides aarch64 work.

Reply:
Just make a mapping with only `intel64` and `arm64`, and throw and error for other unsupported platforms should be fine.

Reply:
Added an extra mapping for intel64.